### PR TITLE
Add count

### PR DESCRIPTION
**Pull request by sophiaso**
Add calculate count for collection or by query.

### CODE REVIEW

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java [hunk 1]
```diff
@@ -406,17 +405,22 @@ private RequestOptions getRequestOptions(PartitionKey key, Integer requestUnit)
...
+    private <T> List<T> executeQuery(@NonNull SqlQuerySpec sqlQuerySpec, boolean isCrossPartition,
                                      @NonNull Class<T> domainClass, String collectionName) {
+        final FeedResponse<Document> feedResponse = executeQuery(sqlQuerySpec, isCrossPartition, collectionName);
+        final List<Document> result = feedResponse.getQueryIterable().toList();
+
+        return result.stream().map(r -> getConverter().read(domainClass, r)).collect(Collectors.toList());
+    }
+
+    private FeedResponse<Document> executeQuery(@NonNull SqlQuerySpec sqlQuerySpec, boolean isCrossPartition,
+                                        String collectionName) {
```
Comment (Incarnation-p-lee) [inline src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java]: should align with last line ?

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java [hunk 2]
```diff
@@ -506,6 +510,42 @@ private boolean isCollectionSupportSortByString(@NonNull DocumentCollection coll
...
+
+        final Object value = feedResponse.getQueryIterable().toList().get(0).getHashMap().get(COUNT_VALUE_KEY);
+        if (value instanceof Integer) {
+            return new Long((Integer)value);
+        } else if (value instanceof Long) {
+            return (Long)value;
+        } else {
+            throw new IllegalStateException("Unexpected value type " + value.getClass() + " of value: " + value);
+        }
+
```
Comment (Incarnation-p-lee) [inline src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java]: should delete this empty line ?

Comment (sophiaso) [reply src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java]: Fixed.

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java (hunk 1 above)
Comment (sophiaso) [reply src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java]: Fixed.

**Pull request merged**